### DISCUSSION TITLE
Make the version of the tree-sitter runtime library available at runtime

### DISCUSSION THREAD

**Discussion by mjambon** [tree-sitter/tree-sitter#1844] [Ideas]
Is there a way to tell which version of the tree-sitter library we're running, at runtime? I didn't find this in [`api.h`](https://github.com/tree-sitter/tree-sitter/blob/master/lib/include/tree_sitter/api.h).

The goal is to be able to log which version is being used, in particular on systems that use dynamic linking. This would allow all sorts of troubleshooting, such as knowing which bugs we can expect to be fixed, performance enhancements, and all the unexpected.

For example, homebrew provides a single tree-sitter package which is [right now at 0.20.6](https://github.com/Homebrew/homebrew-core/blob/b027b16eb71a6b0802875c46243ae1cf223eb774/Formula/tree-sitter.rb#L6). Our own semgrep package (formula) depends on this but I don't see a way to specify a [version constraint on the tree-sitter library](https://github.com/Homebrew/homebrew-core/blob/b027b16eb71a6b0802875c46243ae1cf223eb774/Formula/semgrep.rb#L35).

If I'm correct about the above, then I suggest exposing a `ts_library_version` variable in `api.h`. Its value could be the official release ID if there is one, otherwise it would fall back to the git commit ID. For instance, it could be `0.20.6` or `a882d0b036b0da23cde5caa1325089b4ca4750ba`.

**Comment by ObserverOfTime** [comment]
Semantic version constraints can be handled by CMake/pkgconf or the package manager.
